uuid.go

```diff
@@ -156,6 +158,47 @@ func (u UUID) String() string {
 	return string(buf)
 }
 
+// Format implements fmt.Formatter for UUID values.  It accepts 'h' or 'H' as a format to output only
+// the hex digits of the UUID.  Any other format specifier returns the canonical RFC-4122 representation.
+func (u UUID) Format(f fmt.State, c rune) {
+	//
+	// TODO: dylan.bourque - 2019-02-15
+	// . should we implement %v? if so, what is "a Go-syntax representation" of a UUID when using %#v?
+	//
+	switch c {
+	case 'h', 'H':
```

Comment:
If we are going to do this here, I think the verbs should be `%x` and `%X`, rather than `%h` and `%H`, since the former verbs already mean "print in hexadecimal" in other contexts.

The circumstances are a little unfortunate. Observe this program: https://play.golang.org/p/uRc8V3pnuEa

Unfortunately, out of the box, it seems like `%x` takes the output from `UUID.String` and encodes it in hexadecimal. I'd venture to say (with relative confidence) that nobody relies on this output today, so we could change `%x` to do The Right Thing instead of what it does out now.

That being said, even without adding new code to package uuid, one possible fix to get `%x` to do that is to use `.Bytes()`, as the program shows.

Reply:
I'm fine with %x and %X rather than %h and %H.

You are correct that the default behavior of %x is to take the string RFC-format string returned by `.String()` and hex-encode the byte values of those characters, which I would say is definitively wrong and should be fixed.

My thoughts on using `fmt.Sprintf("%x", u.Bytes())` is the same as for `strings.Replace(u.String(), "-", "", -1)`, namely that there's no logical abstraction.  While correct, either of these amounts to forcing consumers to dig into the internals of the instance.

Reply:
It seems like we have consensus on %x and %X, then. I think that's good enough for this particular point, so I'm going to mark it as resolved.